File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int candidate = 0; 
        int count = 0; 
        for(int n : nums){
            if(count == 0) candidate = n; 
            count += (candidate == n) ? 1 : -1; 
        }
        return candidate;       
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int len = nums.length; 
        Map<Integer,Integer> hashMap = new HashMap<>(); 
        for(int n : nums) hashMap.put(n, hashMap.getOrDefault(n,0) + 1);

        for(Map.Entry<Integer,Integer> e : hashMap.entrySet()){
            if(e.getValue() > (len / 2)){
                System.out.println(e.getValue()); 
                return e.getKey(); 
            }
        }
        return -1; 
    }
}